### PR TITLE
Add command completion

### PR DESCRIPTION
**Pull request by lukas-reineke**
command line completion for `Octo` command

For now just the basic commands.
This could be made smarter to also autocomplete issue number after `Octo issue edit` etc

### CODE REVIEW

File: lua/octo/commands.lua [hunk 1]
```diff
@@ -695,4 +695,31 @@ function M.issue_interactive_action(action, kind)
...
+        table.insert(valid_options, option)
+      end
+    end
+    return valid_options
+  end
+
+  if #parts == 2 then
+    return get_options(command_keys)
+  end
+  if #parts == 3 then
```
Comment (pwntester) [inline lua/octo/commands.lua:716]: elseif?

File: lua/octo/commands.lua [hunk 2]
```diff
@@ -695,4 +695,31 @@ function M.issue_interactive_action(action, kind)
   M.issue_action(action, kind, value)
 end
 
+function M.command_complete(args)
```
Comment (pwntester) [inline lua/octo/commands.lua:698]: not sure if we should move it to `completion.lua` for tidiness. Im ok either way

**Review by pwntester — CHANGES_REQUESTED** [review]

File: lua/octo/commands.lua (hunk 2 above)
Comment (lukas-reineke) [reply lua/octo/commands.lua:698]: Yeah I thought about doing that first, but then `commands` need to be exported. And having that contained in 1 file is kinda nice.

But I'm also fine either way.

File: lua/octo/commands.lua (hunk 1 above)
Comment (lukas-reineke) [reply lua/octo/commands.lua]: `#parts == 2` returns always. But you're right, more explicit to use `elseif`. Pushed a fix

File: lua/octo/commands.lua (hunk 2 above)
Comment (pwntester) [reply lua/octo/commands.lua:698]: ok, lets leave it as is. thanks!

**Comment by pwntester** [comment]
Thanks!!

**Pull request merged**